src/table.js

```diff
@@ -666,3 +762,118 @@ export function __table(source, operations) {
   }
   return source;
 }
+
+function coerceRow(object, types) {
+  let coerced = {};
+  for (var key in object) {
+    const type = types.get(key);
+    const value = object[key];
+    coerced[key] = coerceToType(value, type);
+  }
+  return coerced;
+}
+
+// https://github.com/d3/d3-dsv/issues/45
+const fixTz =
+  new Date("2019-01-01T00:00").getHours() ||
+  new Date("2019-07-01T00:00").getHours();
+
+function initKey() {
+  return {
+    other: 0,
+    boolean: 0,
+    integer: 0,
+    number: 0,
+    date: 0,
+    string: 0,
+    array: 0,
+    object: 0,
+    bigint: 0,
+    buffer: 0
+  };
+}
+
+// We need to show *all* keys present in the array of Objects
+function getAllKeys(rows) {
+  const keys = new Set();
+  for (const row of rows) {
+    // avoid crash if row is null or undefined
+    if (row) {
+      // only enumerable properties
+      for (const key in row) {
+        // only own properties
+        if (Object.prototype.hasOwnProperty.call(row, key)) {
+          // unique properties, in the order they appear
+          keys.add(key);
+        }
+      }
+    }
+  }
+  return Array.from(keys);
+}
+
+export function inferSchema(source) {
+  const allKeys = getAllKeys(source);
```

Comment:
If we already know which columns we are selecting (which we often, but not always, do) we could avoid this and look at operations.select instead.